client/jetpack-cloud/components/missing-payment-notification/index.tsx

```diff
@@ -15,7 +15,7 @@ export default function MissingPaymentNotification() {
 	}
 
 	const latestUnpaidInvoice = partner.keys.reduce< Invoice | null >( ( latestInvoice, key ) => {
```

Comment:
As we discussed, noting that this has two flaws:

1. It doesn't return early.
2. It fails when there are actually `null` values for a key's `latestInvoice`.

Instead we can consider something like this (your code):

```
const firstUnpaidInvoice = partner.keys.find( key => 
    key.latestInvoice && key.latestInvoice.status === 'open'
)?.latestInvoice;
```